### PR TITLE
Cleanup `data` module

### PR DESCRIPTION
**Pull request by cptartur**
## Describe your changes

This PR cleans up `data` module by:

* moving serializers to separate files in serializers module
* removing unused data classes 
* moving data class files (namely `Call`)

## Linked issues

Closes

## Breaking changes

- [x] This issue contains breaking changes

* Import paths changed
* Removed some classes (unused in library)

### CODE REVIEW

**Review by THenry14 — APPROVED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/data/serializers/BlockIdSerializer.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonEncoder
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.put
+
+internal object BlockIdSerializer : KSerializer<BlockId> {
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/serializers/BlockIdSerializer.kt:17]: JvmSynthetic for internals?

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.swmansion.starknet.data.types.transactions
+
+import com.swmansion.starknet.data.selectorFromName
+import com.swmansion.starknet.data.types.Calldata
+import com.swmansion.starknet.data.types.Felt
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class Call(
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: Why was this moved? This is related to execution, it is not a transaction.

File: lib/src/test/kotlin/starknet/provider/ProviderTest.kt [hunk 3]
```diff
@@ -588,7 +587,7 @@ class ProviderTest {
         val payload =
             DeclareTransactionPayload(
                 ContractDefinition(contents),
-                DECLARE_SENDER_ADDRESS,
+                Felt.ONE, // Declare tx version 0 has a sender address of 0x1
```
Comment (Solpatium) [inline lib/src/test/kotlin/starknet/provider/ProviderTest.kt:590]: I don't like replacing properly named constant with a comment. Code > comments.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt [hunk 4]
```diff
@@ -2,78 +2,8 @@ package com.swmansion.starknet.data.types
...
-
-@Serializable
-data class AbiEntry(
-    val name: String,
-    val type: AbiEntryType,
-)
-
-@Serializable
-sealed class AbiElement
-
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt:24]: all that is no longer used due to deploy remove, right?

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt [hunk 5]
```diff
@@ -46,16 +46,6 @@ data class DeployAccountResponse(
...
 
-@Serializable
-data class GetStorageAtResponse(
-    val result: Felt,
-)
-
-data class TransactionFailureReason(
-    val code: String,
-    val errorMessage: String,
-)
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt:57]: There are not used anywhere anymore? 🤔

**Review by Solpatium — CHANGES_REQUESTED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt (hunk 2 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: It's a core part of `InvokeTransactionPayload` so I think it makes sense to keep them in the same module.

File: lib/src/test/kotlin/starknet/provider/ProviderTest.kt (hunk 3 above)
Comment (cptartur) [reply lib/src/test/kotlin/starknet/provider/ProviderTest.kt:590]: Generally I agree, but this constant only made sense in v0 declare transactions. Ideally we'll update our tests to only create declare v1 and this will be replaced entirely.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt (hunk 4 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt:24]: It wasn't used by any part of library even before I removed it. I think these got declared a while ago and finally were never used in any module

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt (hunk 2 above)
Comment (Solpatium) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: But you also use call without **any transaction**, so having it here is misleading.

Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: You are correct about that. I guess the problem is it doesn't make much sense to bundle it with all other data classes in `execution` since it's used in many different places.

Perhaps we should just put `Call.kt` in top level `types` module to indicate that it is it's own separate thing?

Comment (Solpatium) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: I started execution file as container for things related to execution and call is one of them. I see that over time we added things that are less relevant there, like GetBlockTransactionCountPayload. Maybe we should move the others to proper places and leave execution things as they are (Call, Calldata, ExecutionParams etc)?

Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: Perhaps having `Payloads.kt` file for all payloads defined in `Exectution.kt` would make sense. We could then keep call etc. in execution.

Though I don't know if it makes more sense than to extract call (and maybe calldata etc.) to separate file, keeping execution as it is?

Comment (Solpatium) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Call.kt:10]: For me the problem is that payloads don't match execution category, so I would rather assign them to specific categories.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt (hunk 4 above)
Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt:24]: This will actually come in handy in rpc 2.1, so let's leave it. I will remove the unnecessary parts (if there will be some)

File: lib/src/main/kotlin/com/swmansion/starknet/data/serializers/BlockIdSerializer.kt (hunk 1 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/serializers/BlockIdSerializer.kt:17]: You cannot have this annotation on classes/objects unfortunately.

**Review by Solpatium — APPROVED** [review]
👌

**Review by THenry14 — APPROVED** [review]
